### DISCUSSION TITLE
HammerDB supports Redis ?

### DISCUSSION THREAD

**Discussion by anilbommareddy** [TPC-Council/HammerDB#359] [Q&A] [answered]
(1)In HammerDB document says-Full support for **Oracle Database, SQL Server, IBM Db2, MySQL, MariaDB, PostgreSQL** n
(2)But in Configure Schema Build options are shown valid to set Redis.
https://www.hammerdb.com/docs3.3/ch08s03.html and HammerDB-4.4/src/redis
In Point(1) not taking about of Redis  and I saw in few issues on Deprecated Redis.
Here I would like to know is HammerDB support for benchmarking (TPC-C and TPC-H) on Redis.

**Comment by sm-shaw** [comment, marked as answer]
Yes, there is a Redis TPROC-C workload that is deprecated and unsupported.  With Redis being a single-threaded database it did not perform well and as described here it was not fair to compare Redis with relational databases for performance https://redis.io/docs/reference/optimization/benchmarks/
_Redis is, mostly, a single-threaded server from the POV of commands execution (actually modern versions of Redis use threads for different things). It is not designed to benefit from multiple CPU cores. People are supposed to launch several Redis instances to scale out on several cores if needed. It is not really fair to compare one single Redis instance to a multi-threaded data store._

If you wish to run HammerDB against Redis you can enable it by uncommenting its entry in the database.xml file.  However some features may not work, no further work will be done to fix features and it may be removed completely at a future release. 
```
<!--Redis deprecated, uncomment to enable as unsupported
<redis>
	<name>Redis</name>
	<description>Redis</description>
	<prefix>redis</prefix>
	<library>redis</library>
	<workloads>TPROC-C</workloads>
	<commands>redis</commands>
</redis>
```
